### PR TITLE
refactor: change how clients are created and used 

### PR DESCRIPTION
**Pull request by kegsay**
Everything now goes via a `ClientCreationRequest` struct
which controls A) the user, B) multiprocess C) options e.g storage.
This obviates the need for functional `With..` options for creating
clients, which didn't compose well with the way we were using them.

Clearly mark out which functions relate to the test `Instance`
vs which are just true static functions.

This fixes https://github.com/matrix-org/complement-crypto/issues/70

### CODE REVIEW

File: internal/cc/instance.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+	"testing"
+
+	"github.com/matrix-org/complement"
+	"github.com/matrix-org/complement-crypto/internal/api"
+	"github.com/matrix-org/complement-crypto/internal/config"
+	"github.com/matrix-org/complement-crypto/internal/deploy"
+	"github.com/matrix-org/complement/helpers"
+)
+
+// Instance represents a test instance (process
```
Comment (andybalaam) [inline internal/cc/instance.go:16]: Just a missing `)` or is there more to go here?

File: internal/cc/instance.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+func NewInstance(cfg *config.ComplementCrypto) *Instance {
+	return &Instance{
+		ssMutex:                &sync.Mutex{},
+		complementCryptoConfig: cfg,
+	}
+}
+
+func (i *Instance) TestMain(m *testing.M) {
+	//complementCryptoConfig = config.NewComplementCryptoConfigFromEnvVars()
```
Comment (andybalaam) [inline internal/cc/instance.go]: Commented code, looks like it's not needed.

**Review by andybalaam — APPROVED** [review]
API changes look good - much nicer. Skimmed some of the details in the tests. A couple of very minor comments/questions.

File: internal/cc/instance.go (hunk 1 above)
Comment (kegsay) [reply internal/cc/instance.go:16]: Fixed

**Pull request merged**